common/hooks/do-pkg/00-gen-pkg.sh

```diff
@@ -87,6 +94,8 @@ genpkg() {
 		${PKGDESTDIR}
 	rval=$?
 
+	# Unlock binpkg
+	trap - ERR EXIT
```

Comment:
Shouldn't the `trap` reset occur after the lock is removed to avoid any race?

Reply:
Yea I think you're right. I modified the trap function so it checks to see if the lock file exists before deleting it. This allowed me to move the trap to before the lock file is created and the untrap to after the file is removed. This should cover all the bases, including if the lock file fails to be created or removed.